### PR TITLE
Implement SSO + OAUTH2 scenario

### PR DESCRIPTION
**Pull request by Sgitario**
This is the same example as in security Keycloak, but using the OAuth2 quarkus extensions instead of OIDC.

### CODE REVIEW

**Comment by Sgitario** [comment]
JVM failed because:

```
Starting the Java application using /opt/jboss/container/java/run/run-java.sh ...
[0;31mERROR No such file /deployments/security-keycloak-oauth2-1.0.0-SNAPSHOT-runner.jar[0m
```

Thus, depends on https://github.com/quarkus-qe/quarkus-openshift-test-suite/pull/138

**Comment by kshpak** [comment]
Tested via:
`mvn -B -V clean verify -Dquarkus-core-only -pl app-metadata/deployment,app-metadata/runtime,common,security/keycloak-oauth2 -fae -Dts.authenticated-registry -Dinclude.datagrid -Dinclude.serverless`
Test passed. Not approving yet (didn't look at it fully)

**Comment by Sgitario** [comment]
Run tests

File: security/keycloak-oauth2/src/main/resources/application.properties [hunk 1]
```diff
@@ -0,0 +1,9 @@
+# this will be overridden by an env var when deployed to OpenShift,
+# see SecurityKeycloakOpenShiftIT.configureKeycloakUrl
+quarkus.oauth2.introspection-url=http://localhost:8180/auth/realms/test-realm/protocol/openid-connect/token/introspect
+quarkus.oauth2.client-id=test-application-client
+quarkus.oauth2.client-secret=test-application-client-secret
+quarkus.oauth2.role-claim=roles
+
+quarkus.openshift.expose=true
+quarkus.s2i.base-jvm-image=registry.access.redhat.com/openjdk/openjdk-11-rhel7
```
Comment (pjgg) [inline security/keycloak-oauth2/src/main/resources/application.properties:9]: move on to registry.access.redhat.com/ubi8/openjdk-11

**Review by pjgg — CHANGES_REQUESTED** [review]

File: security/keycloak-oauth2/src/main/resources/application.properties (hunk 1 above)
Comment (Sgitario) [reply security/keycloak-oauth2/src/main/resources/application.properties]: Updated

**Comment by Sgitario** [comment]
run tests

**Comment by Sgitario** [comment]
run tests

**Comment by Sgitario** [comment]
run tests

**Review by pjgg — APPROVED** [review]

**Pull request merged**